cloudify_agent/operations.py

```diff
@@ -427,6 +428,7 @@ def _stop_old_agent(new_agent, old_agent, timeout):
 
     _run_script(new_agent, script_url, timeout)
     ctx.logger.info('Old Cloudify agent stopped')
+    utils.update_agent_record(old_agent.get('name'), AgentState.STOPPED)
```

Comment:
there's likely no reason to use `.get` in all those calls - if the name isn't there, then the update will fail anyway. So might as well either fail faster with a keyerror, or have a `if not name: raise ...` inside of `update_agent_record`

Reply:
ok so i'll change to `old_agent['name']`